packages/language-php/grammars/tree-sitter/queries/highlights.scm

```diff
@@ -233,19 +233,56 @@
 
 (class_declaration (name) @entity.name.type.class.php)
 
+(class_declaration (base_clause (name) @entity.other.inherited-class.php))
+(class_declaration (class_interface_clause (name) @entity.other.implemented-interface.php))
+
+; usage or consumption of traits
+(use_declaration (name) @entity.name.type.trait.php)
+; in use lists, "bar" in `Foo::bar` is a method, not a constant
+(use_list
+  (_
+    (class_constant_access_expression (name) @support.other.function.method.php .)
+    (name) @support.other.function.method.php
+  )
+  (#set! capture.final true)
+)
+
+; the "Foo" and "bar" in `Foo::bar`
+(class_constant_access_expression . (name) @support.class.php)
+(class_constant_access_expression (name) @variable.other.property.php .)
+
+; the "Foo" and "$bar" in Foo::$bar
+(scoped_property_access_expression
+  scope: (name) @support.class.php
+  name: (variable_name) @variable.other.property.static.php
+  (#set! capture.final true))
+
 ; Static method calls.
 (scoped_call_expression
+  scope: (name) @support.class.php
   name: (name) @support.other.function.method.static.php)
 
 (member_call_expression
   name: (name) @support.other.function.method.php)
 
-
 ; TRAITS
 ; ======
 
 (trait_declaration (name) @entity.name.type.trait.php)
 
+; INTERFACES
+; ======
+
+(interface_declaration (name) @entity.name.type.interface.php)
+
+; ENUMS
+; =====
+
+(enum_declaration
+  name: (name) @entity.name.type.enum.php
+  (enum_declaration_list
+    (enum_case
+      name: (name) @constant.other.enum.php)))
```

Comment:
This is brand new. I copied this from the Java tree-sitter highlights.

Reply:
Almost! GitHub's suggestion view isn't working right here, but try this instead:

```scm
(enum_declaration
  name: (name) @entity.name.type.enum.php)

(enum_declaration_list
  (enum_case
    name: (name) @constant.other.enum.php))
```

If they're separate queries, then the first capture will get applied even when there aren't any `enum_case`s yet, like if the user is just starting to type the enum block.

Reply:
Oh, yes, that makes sense. tree-sitter is an incremental parser, after all. No need to get in its way!